src/tests/tribol_enzyme_poly_intersect.cpp

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+//-----------------------------------------------------------------------------
+//
+// file: tribol_enzyme_poly_intersect.cpp
+//
+//-----------------------------------------------------------------------------
+
+#include <iostream>
+
+#include "tribol/config.hpp"
+
+#include "gtest/gtest.h"
+
+#ifdef TRIBOL_USE_UMPIRE
+#include "umpire/ResourceManager.hpp"
+#endif
+
+#include "mfem.hpp"
+
+#include "tribol/common/Enzyme.hpp"
+#include "tribol/geom/GeomUtilities.hpp"
+
+namespace tribol {
+
+/**
+ * @brief Test fixture for the Enzyme-based derivatives of intersection polynomial calculations.
+ */
+class EnzymePolyIntersectTest : public testing::Test {
+ protected:
+  static constexpr double delta_{ 1.0e-8 };
+  void SetUp() override {}
+
+  void CheckIntersection( RealT* x1, RealT* x2, int* stencil_dir, RealT fd_tol = delta_, RealT pos_tol = 1.0e-8,
```

Comment:
Consider renaming routine if this checks the intersection or overlap Jacobian contributions.

Reply:
renamed